src/components/SEBankIDSameDeviceButton/SEBankIDSameDeviceButton.tsx

```diff
@@ -0,0 +1,159 @@
+  const [initiated, setInitiated] = useState(false);
+  const {buildAuthorizeUrl, completionStrategy, generatePKCE, domain, handleResponse} = useContext(CriiptoVerifyContext);
+  const {redirectUri} = props;
+
+  const reset = () => {
+    setPKCE(undefined);
+    setLinks(null);
+    setHref(props.href);
+  };
+
+  const handleComplete = useCallback(async (completeUrl: string) => {
+    const required = {pkce};
+    reset();
+
+    const completeResponse = await fetch(completeUrl);
+    if (completeResponse.status >= 400) {
+      setError(await completeResponse.text());
+      return;
+    }
+    
+    const {location}  : {location: string} = await completeResponse.json();
+    if (completionStrategy === 'openidprovider') {
+      window.location.href = location;
+      return;
+    }
+    const url = new URL(location);
+    const params = searchParamsToPOJO(url.searchParams) as AuthorizeResponse;
+
+    await handleResponse(params, {
+      pkce: required.pkce,
+      redirectUri: redirectUri
+    })
+  }, [completionStrategy, pkce]);
+
+  // Mobile visibility scenario
+  usePageVisibility(async () => {
+    if (!links || !mobileOS) return;
+
+    handleComplete(links.completeUrl);
+  }, [links]);
+
+  // Desktop polling scenario
+  useEffect(() => {
+    if (!links) return;
+    if (mobileOS) return;
+    if (!initiated) return;
+    
+    let timeout : string | undefined;
+    const poll = async () => {
+      const response = await fetch(links.pollUrl);
+
+      if (response.status === 202) {
+        setTimeout(poll, 1000);
+        return;
+      } else if (response.status >= 400) {
+        const error = await response.text();
+        setError(error);
+        return;
+      } else {
+        const {targetUrl} = await response.json();
+        await handleComplete(`https://${domain}${targetUrl}`);
+        return;
+      }
+    };
+
+    setTimeout(poll, 1000);
+    return () => {
+      if (timeout) clearTimeout(timeout);
+    };
+  }, [links, initiated]);
+
+  const refresh = useCallback(async () => {
+    console.log('SEBankID: Refresh authorize url');
+    const pkce = await generatePKCE();
+
+    buildAuthorizeUrl({
+      acrValues: 'urn:grn:authn:se:bankid:same-device',
+      loginHint: 'appswitch:browser',
+      responseMode: 'json',
+      pkce,
+      redirectUri
+    }).then(url => {
+      return fetch(url).then(response => response.json() as Promise<Links>);
+    })
+    .then(links => {
+      setPKCE(pkce || undefined);
+      setLinks(links);
+      const redirect = mobileOS === 'ios' ? encodeURIComponent(window.location.href) : 'null';
+      setHref(`${mobileOS ? links.launchLinks.universalLink : links.launchLinks.customFileHandlerUrl}&redirect=${redirect}`);
+    })
+    .catch(console.error);
+  }, [buildAuthorizeUrl, redirectUri]);
+
+  // Generate URL on first button render
+  useEffect(() => {
+    refresh();
+  }, [refresh]);
+
+  // Continously fetch new autostart token if UI is open for a long time
```

Comment:
Could this potentially trigger while the BankID app is in foreground on iOS?

Would explain the error message seen.

Reply:
@sgryt Hmm maybe if the onclick handler doesnt fire on iOS actually, could be it!

Reply:
But could it be the other way around - in the sense that the displayed response is from _before_ the flow is completed in the BankID app ?

Reply:
IIRC, the reason we had to not poll in the HTML UI was because iOs lets JS continue to run when Safari is backgrounded.

Reply:
@sgryt We don't poll per say in the device flows, the intent is simply to refresh links if you're idle on the page so that the order doesn't expire.

We also automatically stop the refreshing as soon as the user lcicks on the button.

Reply:
`setInterval` runs on the tail-end, so doesn't execute untill the first interval has passed.

Reply:
@sgryt Lets debug live with ngrok and slack-huddle tomorrow, will be easier.